drivers/npmx/npmx_driver.c

```diff
@@ -119,14 +119,20 @@ static int int_gpio_interrupt_init(const struct device *dev)
 {
 	struct npmx_data *data = dev->data;
 	const struct npmx_config *config = dev->config;
+
+	if (config->pmic_int_pin != -1) {
```

Comment:
It should me moved to the previous function, the same as here: https://github.com/NordicSemiconductor/npmx-zephyr/pull/77/files#diff-96661c646d9ec2d1a410f047738502496cd688985451a8546a62a3c65fba5919R270

Also clearing interrupts can be skipped.